vault/resource_mongodbatlas_secret_role.go

```diff
@@ -0,0 +1,215 @@
+				ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+			},
+			consts.FieldPath: {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Path where MongoDB Atlas backend is mounted",
+				ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+			},
+			consts.FieldName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the role",
+			},
+			consts.FieldOrganizationID: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "ID for the organization to which the target API Key belongs",
+			},
+			consts.FieldProjectID: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "ID for the project to which the target API Key belongs",
+			},
+			consts.FieldRoles: {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "List of roles that the API Key needs to have",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			consts.FieldIPAddresses: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "IP address to be added to the whitelist for the API key",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			consts.FieldCIDRBlocks: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Whitelist entry in CIDR notation to be added for the API key",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			consts.FieldProjectRoles: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Roles assigned when an org API key is assigned to a project API key",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			consts.FieldTTL: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Duration in seconds after which the issued credential should expire",
+			},
+			consts.FieldMaxTTL: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The maximum allowed lifetime of credentials issued using this role",
+			},
+		},
+	}
+}
+
+func mongodbAtlasSecretRoleCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	backend := d.Get(consts.FieldBackend).(string)
+	name := d.Get(consts.FieldName).(string)
+
+	path := backend + "/roles/" + name
+	log.Printf("[DEBUG] Creating role %q in MongoDB Atlas", name)
+
+	data := map[string]interface{}{}
+	fields := []string{
+		consts.FieldOrganizationID,
+		consts.FieldProjectID,
+		consts.FieldRoles,
+		consts.FieldIPAddresses,
+		consts.FieldCIDRBlocks,
+		consts.FieldProjectRoles,
+		consts.FieldTTL,
+		consts.FieldMaxTTL,
+	}
+	for _, k := range fields {
+		if d.HasChange(k) {
+			data[k] = d.Get(k)
+		}
+	}
+
+	if _, err := client.Logical().Write(path, data); err != nil {
```

Comment:
nit: to match up with the read with context below
```suggestion
	if _, err := client.Logical().WriteWithContext(ctx, path, data); err != nil {
```